manifests/metallb.yaml

```diff
@@ -369,6 +411,11 @@ spec:
       - args:
         - --port=7472
         - --config=config
+        env:
+        - name: METALLB_ML_SECRET_NAME
+          value: memberlist
+        - name: METALLB_DEPLOYMENT
+          value: controller
```

Comment:
I'd rather move these defaults into the code.  Otherwise, these should be documented as more changes in behavior.  I think it's easier to just make the controller do the right thing.

Reply:
`METALLB_DEPLOYMENT` is not `controller` when using helm if I remember correctly, so I would say no for this one

empty / missing `METALLB_ML_SECRET_NAME` disable secret creation, so if I remove it it's disabled which makes more sense than adding `METALLB_ML_SECRET_NAME = ""` to disable it

Let's me document them

Reply:
see https://github.com/metallb/metallb/pull/658/files#diff-829c7adc7abb37fd2f497867860e411c3fb4d73e15b62ce283544ba2f88f02c6R53-R58

Reply:
improved the release notes a bit

Reply:
You should be able to discover the pod identity, right?  Then the pod should have an owner reference to its deployment.

Also there is a SetOwnerReference() helper here we could use, but I don't see controller-runtime in use in metallb, so it would be a new dependency: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#SetControllerReference

Reply:
I remember now, I could from `metadata.uid` have the replication controller, and from the replication controller have the deployment, so it's easier to just go to the deployment, and allow targeted RBAC (only get deployment named controller)

Reply:
I'm not sure I understand.  I think you're saying you found a way to get it without this env var, but you found the env var easier?

Reply:
No, I would still need an env var to get `metadata.uid`, so it's trading an env var for another one (and wider RBAC)